### ISSUE TITLE
[Foundation] Add constant gas fees

### ISSUE THREAD

**Issue by Olshansk** [pokt-network/poktroll#794]
## Objective

Add transaction/gas fees to poktroll.

## Origin Document

https://docs.cosmos.network/main/learn/beginner/gas-fees

https://www.notion.so/buildwithgrove/Proof-submission-cost-Tx-fee-vs-Spend-coins-044bcab72c1d4aef8dba523bd9bdf42d?pvs=4

![Screenshot 2024-09-04 at 3 03 48 PM](https://github.com/user-attachments/assets/93ded5a2-1172-4a2b-a7dc-34cc00e0ed6a)

![Screenshot 2024-09-04 at 3 01 53 PM](https://github.com/user-attachments/assets/2ce14539-7e48-4fd5-bab2-b69849d67c84)

## Goals

- Parity with Morse w.r.t Tx Fees
- Unblock poktscan from deserializing tx fees
- Learning & understanding how cosmos gas fees work

## Deliverables

- [ ] Set a constant tx fee for all transactions
- [ ] Ensure this is part of LocalNet and future TestNets

## Non-goals / Non-deliverables

- Using variable gas meter logic

## General deliverables

- [ ] **Comments**: Add/update TODOs and comments alongside the source code so it is easier to follow.
- [ ] **Testing**: Add new tests (unit and/or E2E) to the test suite.
- [ ] **Makefile**: Add new targets to the Makefile to make the new functionality easier to use.
- [ ] **Documentation**: Update architectural or development READMEs; use [mermaid](https://mermaid-js.github.io/mermaid/) diagrams where appropriate.

---

**Creator**: @Olshansk 
**Co-Owners**: @okdas

**Comment by okdas** [comment]
@Olshansk I've added the gas fees in #768. They're not constant, though. I can look into that to see if constant fees can be easily configured.

**Comment by okdas** [comment]
I flipped the config temporarily in TestNet, and
- faucet broke as it required a fee configuration. Error:
   ```
   err: Error: Broadcasting transaction failed with code 13 (codespace: sdk). Log: insufficient fees; got:  required: 1upokt: insufficient fee
   ```
- All CLI commands that we have in documentation also stopped working due to a similar issue. So we either need to change the documentation or CLI defaults.

Will take a look at both when I start working on the ticket.

**Comment by Olshansk** [comment]
@okdas Once you take a look at it, let's go with `CLI defaults` (if it's possible) and just document the flag in the user guide in one spot.

**Comment by okdas** [comment]
Related doc: https://www.notion.so/buildwithgrove/Proof-submission-cost-Tx-fee-vs-Spend-coins-044bcab72c1d4aef8dba523bd9bdf42d?pvs=4

**Comment by okdas** [comment]
We've decided we are not going to add a constant gas fee. However, we still need to adjust `minimum-gas-prices` on our networks to make sure we don't miss anything gas related. Will prioritize this in preparation for beta testnet and #925.

**Comment by okdas** [comment]
Beta TestNet has been running with the fees turned on, and we're working to add them on LocalNet in #794.

**Comment by okdas** [comment]
No changes. Introducing gas fees to LocalNet is currently not the priority due to a larg~er work on the tests side. We're going to revisit on v0.52+ of cosmos-sdk, as there are some enhancements in the configs we can potentially utilize for fees/default values. 

See #940 for more details.

**Comment by okdas** [comment]
Repurposing this ticket to add gas fees on LocalNet. How the fees work is currently understood and implemented on TestNets.
